app/src/androidTest/java/com/sdp/swiftwallet/uiTest/TransactionActivityTest.java

```diff
@@ -1,35 +1,11 @@
-package com.sdp.swiftwallet.UiTest;
-
-import static androidx.test.espresso.Espresso.onView;
-import static androidx.test.espresso.action.ViewActions.click;
-import static androidx.test.espresso.assertion.ViewAssertions.matches;
-import static androidx.test.espresso.matcher.ViewMatchers.isDisplayed;
-import static androidx.test.espresso.matcher.ViewMatchers.withId;
-
-import android.content.Intent;
-
-import androidx.test.core.app.ActivityScenario;
-import androidx.test.core.app.ApplicationProvider;
-import androidx.test.ext.junit.runners.AndroidJUnit4;
-
-import com.sdp.cryptowalletapp.R;
-import com.sdp.swiftwallet.SwiftWalletApp;
-import com.sdp.swiftwallet.domain.model.Currency;
-import com.sdp.swiftwallet.domain.model.Transaction;
-import com.sdp.swiftwallet.domain.repository.TransactionHistoryProducer;
-import com.sdp.swiftwallet.domain.repository.TransactionHistorySubscriber;
-import com.sdp.swiftwallet.presentation.transactions.TransactionActivity;
-
-import org.junit.Before;
-import org.junit.Test;
-import org.junit.runner.RunWith;
-
-import java.util.ArrayList;
-import java.util.List;
-
+package com.sdp.swiftwallet.uiTest;
 
+/**
+@HiltAndroidTest
 @RunWith(AndroidJUnit4.class)
 public class TransactionActivityTest {
```

Comment:
We might want to try for next time to have the custom hilt runner launch an implementation of the same class as the SwiftWalletApp. Otherwise the entire purpose of the app "context" is lost.

Reply:
Yep that's the plan ! I think we can put it as a task